mentat/conversation.py

```diff
@@ -52,7 +55,25 @@ def get_model_response(self, config: ConfigManager) -> (str, list[CodeChange]):
         messages = self.messages.copy()
 
         code_message = self.code_file_manager.get_code_message()
-        messages.append({"role": "system", "content": code_message})
+        system_message = code_message
+
+        if self.code_map:
+            code_map_message = self.code_map.get_message()
+            if code_map_message is not None:
+                code_map_message_token_count = count_tokens(code_map_message)
+                prompt_token_count = 0
+                prompt_token_count += count_tokens(code_message)
+                for message in messages:
+                    prompt_token_count += count_tokens(message["content"])
```

Comment:
perhaps we should make `self.messages` more than just a list of messages, but instead also store token counts for each message and/or a total, so we don't recalculate again and again?

don't have to make that change as part of this PR, we can make a separate issue for that

Reply:
I'm wondering how much of a performance increase this would be? I don't think it's much work to set this up though so it's worth it to see if it helps a good amount.

Reply:
Yeah it may not really matter, but if done cleanly would be nice. Doesn't have to be this PR though